src/plugins/unified_search/public/filter_bar/filter_editor/value_input_type.tsx

```diff
@@ -26,6 +26,16 @@ interface Props {
 }
 
 class ValueInputTypeUI extends Component<Props> {
+  private getValueForNumberField = (value?: string | number) => {
+    if (typeof value === 'string') {
+      const parsedValue = parseFloat(value);
+      if (!isNaN(parsedValue)) {
+        return parsedValue;
+      }
+      return value;
```

Comment:
nit, this seems slightly clearer to me:
```suggestion
      return isNaN(parseValue) ? value : parsedValue;
```